ui/src/components/SessionProvider.tsx

```diff
@@ -27,8 +27,30 @@ export default function SessionProvider({
   const [session, setSession, clearSession] = useLocalStorage('session', null);
 
   useEffect(() => {
+    const clearSessionIfNecessary = async () => {
+      const config = await getConfig();
+      if (session && session.required === false) {
+        if (config.authentication.required) {
+          clearSession();
+        }
+      }
+
+      if (session && session.required === true) {
+        if (!config.authentication.required) {
+          clearSession();
+        }
+      }
```

Comment:
```suggestion
      if (session && session.required === config.authentication.required) {
          clearSession();
      }
```

is ^ functionally the same?